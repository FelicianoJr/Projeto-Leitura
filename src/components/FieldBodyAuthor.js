import React from "react";
import { Field } from "redux-form";

const FieldBodyAuthor = () => {
  return (
    <div>
      
        <div className="form-group">
          <label >Autor</label>
          <Field
            name="author"
            className="form-control"
            id="author"
            component="input"
            type="text"
          />
        </div>
        <div className="form-group">
          <label >Descrição</label>
          <Field
            name="body"
            className="form-control"
            id="body"
            component="textarea"
            type="text"
          />
      </div>
    </div>
  );
};

export default FieldBodyAuthor;
