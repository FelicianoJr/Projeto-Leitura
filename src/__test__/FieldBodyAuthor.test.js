import React from "react";
import { shallow } from "enzyme";
import FieldBodyAuthor from "../components/FieldBodyAuthor";

const setup = () => {
  const component = shallow(<FieldBodyAuthor />);

  return {
    component: component,
    formGroup: component.find(".form-group"),
    label: component.find("label"),
    input: component.find("Field")
  };
};

describe("<FieldBodyAuthor/>", () => {
  it("should rende  <Form>", () => {
    const { formGroup } = setup();
    expect(formGroup.length).toEqual(2);
  });

  it("should rende  <Label> ", () => {
    const { label } = setup();
    expect(label.at(0).text()).toEqual("Autor");
    expect(label.at(1).text()).toEqual("Descrição");
  });

  it("should rende  <Label> ", () => {
    const { input } = setup();
    expect(input.length).toEqual(2);
  });

  it("should component Snapshot", () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
