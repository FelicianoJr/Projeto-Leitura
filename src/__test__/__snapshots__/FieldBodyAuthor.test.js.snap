// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldBodyAuthor/> should component Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FieldBodyAuthor />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="form-group"
>
          <label>
                    Autor
          </label>
          <Field
                    className="form-control"
                    component="input"
                    id="author"
                    name="author"
                    type="text"
          />
</div>,
        <div
          className="form-group"
>
          <label>
                    Descrição
          </label>
          <Field
                    className="form-control"
                    component="textarea"
                    id="body"
                    name="body"
                    type="text"
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Autor
</label>,
            <Field
              className="form-control"
              component="input"
              id="author"
              name="author"
              type="text"
/>,
          ],
          "className": "form-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Autor",
            },
            "ref": null,
            "rendered": "Autor",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "form-control",
              "component": "input",
              "id": "author",
              "name": "author",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Descrição
</label>,
            <Field
              className="form-control"
              component="textarea"
              id="body"
              name="body"
              type="text"
/>,
          ],
          "className": "form-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Descrição",
            },
            "ref": null,
            "rendered": "Descrição",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "form-control",
              "component": "textarea",
              "id": "body",
              "name": "body",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="form-group"
>
            <label>
                        Autor
            </label>
            <Field
                        className="form-control"
                        component="input"
                        id="author"
                        name="author"
                        type="text"
            />
</div>,
          <div
            className="form-group"
>
            <label>
                        Descrição
            </label>
            <Field
                        className="form-control"
                        component="textarea"
                        id="body"
                        name="body"
                        type="text"
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Autor
</label>,
              <Field
                className="form-control"
                component="input"
                id="author"
                name="author"
                type="text"
/>,
            ],
            "className": "form-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Autor",
              },
              "ref": null,
              "rendered": "Autor",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "form-control",
                "component": "input",
                "id": "author",
                "name": "author",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Descrição
</label>,
              <Field
                className="form-control"
                component="textarea"
                id="body"
                name="body"
                type="text"
/>,
            ],
            "className": "form-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Descrição",
              },
              "ref": null,
              "rendered": "Descrição",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "form-control",
                "component": "textarea",
                "id": "body",
                "name": "body",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
